### ISSUE TITLE
infinite with xcode 9 GM

### ISSUE THREAD

**Issue by bricecesarin** [Flinesoft/BartyCrouch#66]
Hello,

I tried my project with the new xcode 9 GM today, the BartyCrouch script never stops. 
So I tried launching command : 
`bartycrouch interfaces -p [My Project Path] -v`
The script is working for some files and then stops for one.
I tried launching with the source file and found that the problem comes from the file : `Commander.swift` , the line `let outdata = outpipe.fileHandleForReading.readDataToEndOfFile()` never stops.

I reproduced the same issue with a second project.

My XCode version is : 9.0 (9A235).

**Comment by mathieuedet** [comment]
Hello bricecesarin ! I've exactly the same issue with my project with the last Xcode version.

**Comment by donnellyk** [comment]
I am also seeing this. 

The `bartycrouch interfaces` command hangs indefinitely, but if I run `/usr/bin/ibtool --export-strings-file` directly, the strings are still exported, though with a bunch of warnings: `2017-09-13 13:37:43.724 ibtoold[3253:4694192] WARNING: Unhandled destination metrics: (null)`

xcode-select points to the GM right now (though I have the old Xcode still installed)
ibtool --version:
```
<key>bundle-version</key>
<string>13196</string>
<key>short-bundle-version</key>
<string>9.0</string>
```

The nib it hangs on does have autolayout errors and if I delete the views causing the errors, bartycrouch runs fine. (Right now, I can't figure out a way to _resolve_ the errors, IB is being weird).

I'm trying to create a nib from scratch that exhibits the same problem. I'll share here if I'm successful...

**Comment by donnellyk** [comment]
If I had to make a guess, it seems like this new version of ibtool isn't given an end-of-file indicator to `standardOutput` when there are errors (in this case, warnings).

It looks like commenting out `let outdata = outpipe.fileHandleForReading.readDataToEndOfFile()
` (so just _ignoring_ the standard output) and the rest of the command runs fine, the string files get made. 

Not that this is an _actual_ solution or anything, just experimenting.

**Comment by bricecesarin** [comment]
when I launch ibtool command from command line : 
`/usr/bin/ibtool --export-strings-file [myTmpStrings] [MyXib]`
the warnings : 
`WARNING: Unhandled destination metrics: (null)`
are appearing for all xib (event those which works).
It seems like the output are not different from the working xibs and the no-working xib.

**Comment by bricecesarin** [comment]
The only output difference is : 
`/usr/bin/ibtool --export-strings-file [myTmpStrings] [MyWorkingXib]` : less than 250 warnings
and
`/usr/bin/ibtool --export-strings-file [myTmpStrings] [MyNoWorkingXib]` : more than 250 warnings
Same thing with my 2 projects

**Comment by bmironer** [comment]
I got a fix. 
Simply reading stderr before stdout solves it.
Pull request incoming after a few more tests.

**Comment by Dschee** [comment]
Fixed via #71.
